packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -270,14 +273,13 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
 
   // Call user-defined onChange when the value actually changes
   React.useEffect(() => {
-    if (selectedIndex != undefined && selectedIndex >= 0) {
-      const value = options[selectedIndex]?.value;
-      if (value === valueProp) {
-        return;
-      }
-      onChangeProp.current?.(value);
+    if (firstRenderRef.current) {
+      firstRenderRef.current = false;
+      return;
```

Comment:
Do you know why this fixes the issue? Can you add a comment then?

Also add a test

Reply:
This solves only issue where useEffect is called on first render. It's called because combobox value is set (undefined or actual value). And we want to allow undefined value for users to reset combobox.
Will add comments.

Reply:
Added test to programmatically reset value.